media/server/main/source/MediaPipelineServerInternal.cpp

```diff
@@ -587,12 +588,43 @@ bool MediaPipelineServerInternal::haveDataInternal(MediaSourceStatus status, uin
         return true;
     }
     m_activeRequests->erase(needDataRequestId);
+
+    int &counter = m_noAvailableSamplesCounter[mediaSourceType];
```

Comment:
move this line into if (status == MediaSourceStatus::NO_AVAILABLE_SAMPLES) scope